api/infolabels.go

```diff
@@ -0,0 +1,190 @@
+        "ListItem.Tagline",
+        "ListItem.Year",
+        "ListItem.Trailer",
+        "ListItem.Studio",
+        "ListItem.MPAA",
+        "ListItem.Genre",
+        "ListItem.Mediatype",
+        "ListItem.Writer",
+        "ListItem.Director",
+        "ListItem.Rating",
+        "ListItem.Votes",
+        "ListItem.IMDBNumber",
+        "ListItem.Code",
+        "ListItem.ArtFanart",
+        "ListItem.ArtBanner",
+        "ListItem.ArtPoster",
+        "ListItem.ArtTvshowPoster",
+    }
+)
+
+func GetEpisodeInfoLabels(show *tmdb.Show, episode *tmdb.Episode) *xbmc.ListItem {
+    item := episode.ToListItem(show)
+    libraryItem := FindEpisodeInLibrary(show, episode)
+    if libraryItem != nil {
+        item.Info.DBID = libraryItem.ID
+    }
+
+    return processItem(item)    
+}
+
+func GetShowInfoLabels(show *tmdb.Show) *xbmc.ListItem {
+    item := show.ToListItem()
+
+    return processItem(item)
+}
+
+func GetMovieInfoLabels(movie *tmdb.Movie) *xbmc.ListItem {
+    item := movie.ToListItem()
+    libraryItem := FindMovieInLibrary(movie)
+    if libraryItem != nil {
+        item.Info.DBID = libraryItem.ID
+    }
+
+    return processItem(item)
+}
+
+func processItem(item *xbmc.ListItem) *xbmc.ListItem {
+    saveEncoded(encodeItem(item))
+
+    return item
+}
+
+func saveEncoded(encoded string) {
+    xbmc.SetWindowProperty("ListItem.Encoded", encoded)
+}
+
+func GetStoredInfoLabels() string {
+    if dbid := xbmc.InfoLabel("ListItem.DBID"); len(dbid) > 0 {
+        labels := xbmc.InfoLabels(infoLabels...)
+
+        listItemLabels := make(map[string]string, len(labels))
+        for k, v := range labels {
+            key := strings.Replace(k, "ListItem.", "", 1)
+            listItemLabels[key] = v
+        }
+
+        encoded := convertMapToJsonString(listItemLabels)
+        saveEncoded(encoded)
+
+        return encoded        
+    } else if encoded := xbmc.GetWindowProperty("ListItem.Encoded"); len(encoded) > 0 {
+        return encoded
+    } else {
+        return "{}"
+    }
+}
+
+func encodeItem(item *xbmc.ListItem) string {
+    data, _ := json.Marshal(item)
+
+    return string(data)
+}
+
+func convertMapToJsonString(obj map[string]string) string {
+    b, _ := json.Marshal(obj)
+    
+    return string(b)
+}
+
+func convertItemToMap(item *xbmc.ListItem) map[string]string {
+    b, _ := json.Marshal(item)
+    m := make(map[string]string)
+    json.Unmarshal(b, &m)
+
+    return m
+}
+
+
+func InfoLabelsStored(btService *bittorrent.BTService) gin.HandlerFunc {
+    return func(ctx *gin.Context) {
+        labels := GetStoredInfoLabels()
+
+        ctx.Writer.Header().Set("Access-Control-Allow-Origin", "*")
+        ctx.String(200, labels)
+    }
+}
+
+func InfoLabelsShow(btService *bittorrent.BTService) gin.HandlerFunc {
+    return func(ctx *gin.Context) {
+        tmdbId := ctx.Params.ByName("showId")
+
+        show := tmdb.GetShowById(tmdbId, "")
+        labels := GetShowInfoLabels(show)
```

Comment:
`GetShowInfoLabels` is only used here, no need for it to be exported, but might as well just put its code here instead and get rid of it. Same for the other ones.